diff -up thunderbird-52.0/mail/installer/Makefile.in.fix-dupes thunderbird-52.0/mail/installer/Makefile.in
--- thunderbird-52.0/mail/installer/Makefile.in.fix-dupes	2017-04-04 13:06:17.414621079 +0200
+++ thunderbird-52.0/mail/installer/Makefile.in	2017-04-04 13:06:24.397611123 +0200
@@ -15,6 +15,7 @@ ifndef SYSTEM_LIBXUL
 MOZ_PKG_FATAL_WARNINGS = 1
 endif
 MOZ_PKG_DUPEFLAGS = \
+	-w \
 	-f $(srcdir)/allowed-dupes.mn \
 	-f $(MOZILLA_DIR)/browser/installer/allowed-dupes.mn \
 	$(NULL)
