diff -up thunderbird-60.3.0/toolkit/components/remote/nsRemoteService.cpp.old thunderbird-60.3.0/toolkit/components/remote/nsRemoteService.cpp
--- thunderbird-60.3.0/toolkit/components/remote/nsRemoteService.cpp.old	2018-11-06 11:25:46.634929894 +0100
+++ thunderbird-60.3.0/toolkit/components/remote/nsRemoteService.cpp	2018-11-06 11:21:49.617828440 +0100
@@ -34,7 +34,7 @@ NS_IMPL_ISUPPORTS(nsRemoteService,
 NS_IMETHODIMP
 nsRemoteService::Startup(const char* aAppName, const char* aProfileName)
 {
-#if defined(MOZ_ENABLE_DBUS)
+#if 0
     nsresult rv;
     mDBusRemoteService = new nsDBusRemoteService();
     rv = mDBusRemoteService->Startup(aAppName, aProfileName);
