Index: widget/src/gtk2/nsWindow.cpp
===================================================================
RCS file: /cvsroot/mozilla/widget/src/gtk2/nsWindow.cpp,v
retrieving revision 1.185
diff -u -8 -p -r1.185 nsWindow.cpp
--- widget/src/gtk2/nsWindow.cpp	20 Sep 2006 19:16:19 -0000	1.185
+++ widget/src/gtk2/nsWindow.cpp	22 Sep 2006 05:07:04 -0000
@@ -2859,19 +2859,16 @@ nsWindow::NativeCreate(nsIWidget        
         g_signal_connect(G_OBJECT(mShell), "window_state_event",
                          G_CALLBACK(window_state_event_cb), NULL);
 
         GtkSettings* default_settings = gtk_settings_get_default();
         g_signal_connect_after(default_settings,
                                "notify::gtk-theme-name",
                                G_CALLBACK(theme_changed_cb), this);
         g_signal_connect_after(default_settings,
-                               "notify::gtk-key-theme-name",
-                               G_CALLBACK(theme_changed_cb), this);
-        g_signal_connect_after(default_settings,
                                "notify::gtk-font-name",
                                G_CALLBACK(theme_changed_cb), this);
     }
 
     if (mContainer) {
         g_signal_connect_after(G_OBJECT(mContainer), "size_allocate",
                                G_CALLBACK(size_allocate_cb), NULL);
         g_signal_connect(G_OBJECT(mContainer), "expose_event",
