See https://bugzilla.mozilla.org/show_bug.cgi?id=367203

Index: mozilla/widget/src/gtk2/nsDragService.cpp
===================================================================
RCS file: /cvsroot/mozilla/widget/src/gtk2/nsDragService.cpp,v
retrieving revision 1.9.10.1
diff -d -u -p -r1.9.10.1 nsDragService.cpp
--- mozilla/widget/src/gtk2/nsDragService.cpp	22 Jun 2006 21:37:45 -0000	1.9.10.1
+++ mozilla/widget/src/gtk2/nsDragService.cpp	31 Jan 2007 04:27:43 -0000
@@ -799,7 +799,6 @@ nsDragService::IsTargetContextList(void)
 void
 nsDragService::GetTargetDragData(GdkAtom aFlavor)
 {
-    gtk_grab_add(mHiddenWidget);
     PR_LOG(sDragLm, PR_LOG_DEBUG, ("getting data flavor %d\n", aFlavor));
     PR_LOG(sDragLm, PR_LOG_DEBUG, ("mLastWidget is %p and mLastContext is %p\n",
                                    mTargetWidget, mTargetDragContext));
@@ -817,7 +816,6 @@ nsDragService::GetTargetDragData(GdkAtom
         gtk_main_iteration();
     }
     PR_LOG(sDragLm, PR_LOG_DEBUG, ("finished inner iteration\n"));
-    gtk_grab_remove(mHiddenWidget);
 }
 
 void
