diff -up mozilla/configure.in.system_nss mozilla/configure.in
--- mozilla/configure.in.system_nss	2014-01-30 23:56:48.000000000 +0100
+++ mozilla/configure.in	2014-01-31 08:56:18.228879795 +0100
@@ -3944,7 +3944,7 @@ MOZ_ARG_WITH_BOOL(system-nss,
     _USE_SYSTEM_NSS=1 )
 
 if test -n "$_USE_SYSTEM_NSS"; then
-    AM_PATH_NSS(3.16.2, [MOZ_NATIVE_NSS=1], [AC_MSG_ERROR([you don't have NSS installed or your version is too old])])
+    AM_PATH_NSS(3.16.1, [MOZ_NATIVE_NSS=1], [AC_MSG_ERROR([you don't have NSS installed or your version is too old])])
 fi
 
 if test -n "$MOZ_NATIVE_NSS"; then
diff -up mozilla/configure.system_nss mozilla/configure
--- mozilla/configure.system_nss	2014-01-31 08:56:18.000000000 +0100
+++ mozilla/configure	2014-01-31 09:14:03.252049304 +0100
@@ -15224,7 +15224,7 @@ else
   echo "$ac_t""no" 1>&6
 fi
 
-	min_nss_version=3.16.2
+	min_nss_version=3.16.1
 	echo $ac_n "checking for NSS - version >= $min_nss_version""... $ac_c" 1>&6
 echo "configure:15230: checking for NSS - version >= $min_nss_version" >&5
 
