diff --git a/gfx/angle/checkout/src/compiler/translator/tree_util/IntermTraverse.cpp b/gfx/angle/checkout/src/compiler/translator/tree_util/IntermTraverse.cpp
index c38baa12bffa8..2858a7bb4d319 100644
--- a/gfx/angle/checkout/src/compiler/translator/tree_util/IntermTraverse.cpp
+++ b/gfx/angle/checkout/src/compiler/translator/tree_util/IntermTraverse.cpp
@@ -45,16 +45,20 @@ void TIntermTraverser::traverse(T *node)
             ++childIndex;
         }
 
         if (visit && postVisit)
             node->visit(PostVisit, this);
     }
 }
 
+// Instantiate template for RewriteAtomicFunctionExpressions.
+template
+void TIntermTraverser::traverse(TIntermNode *);
+
 void TIntermNode::traverse(TIntermTraverser *it)
 {
     it->traverse(this);
 }
 
 void TIntermSymbol::traverse(TIntermTraverser *it)
 {
     TIntermTraverser::ScopedNodeInTraversalPath addToPath(it, this);
