diff -up thunderbird-60.6.1/js/src/util/NativeStack.cpp.1533969 thunderbird-60.6.1/js/src/util/NativeStack.cpp
diff -up thunderbird-60.6.1/tools/profiler/core/platform.h.1533969 thunderbird-60.6.1/tools/profiler/core/platform.h
--- thunderbird-60.6.1/tools/profiler/core/platform.h.1533969	2019-03-26 12:51:50.138988424 +0100
+++ thunderbird-60.6.1/tools/profiler/core/platform.h	2019-03-26 12:54:57.576579732 +0100
@@ -47,11 +47,11 @@
 #if defined(__GLIBC__)
 #include <unistd.h>
 #include <sys/syscall.h>
-static inline pid_t gettid() { return (pid_t)syscall(SYS_gettid); }
+#  define gettid() static_cast<pid_t>(syscall(SYS_gettid))
 #elif defined(GP_OS_darwin)
 #include <unistd.h>
 #include <sys/syscall.h>
-static inline pid_t gettid() { return (pid_t)syscall(SYS_thread_selfid); }
+#  define gettid() static_cast<pid_t>(syscall(SYS_thread_selfid))
 #elif defined(GP_OS_android)
 #include <unistd.h>
 #elif defined(GP_OS_windows)
