diff -up firefox-69.0/toolkit/xre/nsAppRunner.cpp.old-version-check firefox-69.0/toolkit/xre/nsAppRunner.cpp
--- firefox-69.0/toolkit/xre/nsAppRunner.cpp.old-version-check	2019-10-02 16:58:38.018077162 +0200
+++ firefox-69.0/toolkit/xre/nsAppRunner.cpp	2019-10-02 16:59:57.955339034 +0200
@@ -2453,13 +2453,9 @@ int32_t CompareCompatVersions(const nsAC
     return result;
   }
 
-  // Fall back to build ID comparison.
-  result = CompareBuildIDs(oldAppBuildID, newAppBuildID);
-  if (result != 0) {
-    return result;
-  }
-
-  return CompareBuildIDs(oldPlatformBuildID, newPlatformBuildID);
+  // Don't compare build ids because the build time can be older on the
+  // newer distribution package.
+  return 0;
 }
 
 /**
